src-built-in/components/workspaceManagementMenu/src/stores/workspaceManagementMenuStore.js

```diff
@@ -350,11 +363,12 @@ Actions = {
 		 * Actually perform the switch. Happens after we ask the user what they want.
 		 *
 		 */
-		function switchWorkspace() {
+		function switchWorkspace(callback) {
 			FSBL.Clients.WorkspaceClient.switchTo({
 				name: name
 			}, () => {
-				Actions.setIsSwitchingWorkspaces(false);
+					Actions.setIsSwitchingWorkspaces(false);
+					callback();
```

Comment:
The callback wasn't needed before to call this, now it's being added as a non-optional parameter. If this is to be added it should be optional and not fail if unprovided.

If it is to be required, we need to make sure any calls to it we make are up-to-date. Since it is on the front-end though, this does expose a breakage concern.

Reply:
See the above comment.  It was needed before -- this just never worked correctly.  The callback is a non-optional parameter since passed into async.waterfall().